core/schema/cache.go

```diff
@@ -35,10 +37,26 @@ type cacheArgs struct {
 	Key string
 }
 
-func (s *cacheSchema) cacheVolume(ctx context.Context, parent *core.Query, args cacheArgs) (*core.CacheVolume, error) {
-	// TODO(vito): inject some sort of scope/session/project/user derived value
-	// here instead of a static value
-	//
-	// we have to inject something so we can tell it's a valid ID
-	return core.NewCache(args.Key), nil
+func (s *cacheSchema) cacheVolume(ctx context.Context, parent dagql.Instance[*core.Query], args cacheArgs) (dagql.Instance[*core.CacheVolume], error) {
+	var inst dagql.Instance[*core.CacheVolume]
+
+	m, err := parent.Self.Server.CurrentModule(ctx)
+	if err != nil && !errors.Is(err, core.ErrNoCurrentModule) {
```

Comment:
will this continue working in non-module pipelines? @rajatjindal is currently testing this now.

Reply:
Hi @marcosnils 

I verified that this works with non-module pipelines. I used following code to verify this:

```
package main

import (
	"context"
	"fmt"
	"os"

	"dagger.io/dagger/dag"
)

func main() {
	ctx := context.TODO()

	vol := dag.CacheVolume("volume-name")
	outout, err := dag.Container().From("alpine:latest").
		WithMountedCache("/foo", vol).
		WithExec([]string{"sh", "-c", "touch /foo/bar.txt"}).
		WithExec([]string{"sh", "-c", "echo 'hello'"}).
		Stdout(ctx)

	if err != nil {
		fmt.Printf("ERROR: %v\n", err)
		os.Exit(1)
	}

	fmt.Println(outout)
}

```